run/jenkins.py

```diff
@@ -0,0 +1,282 @@
+
+    patch_wpt(args['wptd_path'], args['wpt_path'], platform)
+
+    if platform.get('sauce'):
+        command = [
+            './wpt', 'run', product,
+            '--sauce-platform=%s' % platform['os_name'],
+            '--sauce-key=%s' % args['sauce_key'],
+            '--sauce-user=%s' % args['sauce_user'],
+            '--sauce-tunnel-id=%s' % SAUCE_TUNNEL_ID,
+            '--no-restart-on-unexpected',
+            '--processes=2',
+            '--run-by-dir=3',
+            '--log-mach=-',
+            '--log-wptreport=%s' % LOCAL_REPORT_FILEPATH,
+            '--install-fonts'
+        ]
+        if args['run_path']:
+            command.insert(3, args['run_path'])
+    else:
+        command = [
+            'xvfb-run', '--auto-servernum',
+            './wpt', 'run',
+            platform['browser_name'],
+            '--install-fonts',
+            '--install-browser',
+            '--yes',
+            '--log-mach=%s' % LOCAL_LOG_FILEPATH,
+            '--log-wptreport=%s' % LOCAL_REPORT_FILEPATH,
+        ]
+        if args['run_path']:
+            command.insert(5, args['run_path'])
+
+    if args['jenkins_job_name'] and '_of_' in args['jenkins_job_name']:
+        # Expects a job name like chunk_2_of_4
+        chunks = re.findall(r'(\d+)_of_(\d+)', args['jenkins_job_name'])
+        chunk_num, total_chunks = chunks[0]
+        command.extend([
+            '--this-chunk=%s' % chunk_num,
+            '--total-chunks=%s' % total_chunks,
+        ])
+
+    return_code = subprocess.call(command, cwd=args['wpt_path'])
+
+    print('==================================================')
+    print('Finished WPT run')
+    print('Return code from wptrunner: %s' % return_code)
+
+    with open(LOCAL_REPORT_FILEPATH) as f:
+        report = json.load(f)
+
+    assert len(report['results']) > 0, (
+        '0 test results, something went wrong, stopping.')
+
+    summary = report_to_summary(report)
+
+    print('==================================================')
+    print('Writing summary.json.gz to local filesystem')
+    write_gzip_json(LOCAL_SUMMARY_GZ_FILEPATH, summary)
+    print('Wrote file %s' % LOCAL_SUMMARY_GZ_FILEPATH)
+
+    print('==================================================')
+    print('Writing individual result files to local filesystem')
+    for result in report['results']:
+        test_file = result['test']
+        filepath = '%s%s' % (GS_RESULTS_FILEPATH_BASE, test_file)
+        write_gzip_json(filepath, result)
+        print('Wrote file %s' % filepath)
+
+    if not (args['prod_run'] or args['prod_wet_run']):
+        print('==================================================')
+        print('Stopping here (pass PROD_RUN env var to upload results).')
+        return
+
+    print('==================================================')
+    print('Uploading results to gs://%s' % GS_RESULTS_BUCKET)
+
+    # TODO: change this from rsync to cp
```

Comment:
Is there context here? A bug to track? Why are we not already using `cp`?

Reply:
Bug: #80 

We are not already using `cp` because I am an idiot

Reply:
Glad it's tracked! Thanks.

Reply:
Added the bug number to the TODO